driver/sql/projector_notification.go

```diff
@@ -135,25 +132,28 @@ func (s *NotificationProjector) project(
 		}
 	}()
 
+	// Wrap the eventStream
+	handlerStream := &eventStreamHandlerIterator{
+		stream:   eventStream,
+		handlers: s.handlers,
+		resolver: s.resolver,
+	}
+
 	// project event stream
-	if err := s.projectStream(ctx, conn, notification, rawState, eventStream); err != nil {
+	if err := s.projectStream(ctx, transaction, handlerStream); err != nil {
 		return err
 	}
 
 	return eventStream.Close()
 }
-
-// projectStream will project the events in the event stream and persist the state after the projection
-func (s *NotificationProjector) projectStream(
+func (s *notificationProjector) projectStream(
 	ctx context.Context,
-	conn driverSQL.Execer,
-	notification *driverSQL.ProjectionNotification,
-	rawState *driverSQL.ProjectionRawState,
-	stream goengine.EventStream,
+	tx ProjectorTransaction,
+	stream *eventStreamHandlerIterator,
 ) error {
 	var (
-		state         driverSQL.ProjectionState
-		stateAcquired bool
+		err   error
```

Comment:
Maybe we can move this to the return of the function? So ultimately we use 3 less lines here.

```
func (s *notificationProjector) projectStream(...) (err error) {
...
}
```